auth/snippets.go

```diff
@@ -54,6 +56,64 @@ func adc() {
 		log.Fatal(err)
 	}
 
-	// [END auth_cloud_implicit]
 	_ = kmsService
 }
+
+// [END auth_cloud_implicit]
+
+// [START auth_cloud_explicit]
+
+// explicit reads credentials from the specified path.
+func explicit(jsonPath, projectID string) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx, option.WithCredentialsFile(jsonPath))
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Buckets:")
+	it := client.Buckets(ctx, projectID)
+	for {
+		battrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(battrs.Name)
+	}
+}
+
+// [END auth_cloud_explicit]
+
+// [START auth_cloud_explicit_compute_engine]
+// [START auth_cloud_explicit_app_engine]
+
+// explicitDefault finds the default credentials.
+func explicitDefault(projectID string) {
+	ctx := context.Background()
+
+	creds, err := google.FindDefaultCredentials(ctx, storage.ScopeReadOnly)
```

Comment:
Why would you need to do this explicitly? `storage.NewClient` with no options will do exactly this, I believe. (I think you answered this in your IM to me, but I didn't understand it.)

Reply:
I'm not aware of why you would need `creds` directly. This happens to use the same storage API to demonstrate, but this isn't the recommended way to do it. Maybe we should add a comment like:

> It is very uncommon to need to explicitly get the default credentials. Most of the time, client libraries can use Application Default Credentials without having to pass the credentials in directly. See implicit above.

What do you think? I think it's better to have a Go entry in the doc, even if it says this is not recommended.

Reply:
Oh, I get it, this is part of a multi-language piece of advice. Yes, please add that comment.

Reply:
Done.